tests/integration/test_kms.py

```diff
@@ -1222,3 +1175,74 @@ def test_cross_accounts_access(self, aws_client, aws_client_factory, kms_create_
         assert client.verify_mac(
             KeyId=key_arn_3, Message=message, MacAlgorithm="HMAC_SHA_512", Mac=mac
         )["MacValid"]
+
+
+class TestKMSGenerateKeys:
+    @pytest.fixture(autouse=True)
+    def generate_key_transformers(self, snapshot):
+        snapshot.add_transformer(snapshot.transform.resource_name())
+
+    @pytest.mark.aws_validated
+    def test_generate_data_key_pair_without_plaintext(self, kms_key, aws_client, snapshot):
+        snapshot.add_transformer(
+            snapshot.transform.key_value("PrivateKeyCiphertextBlob", reference_replacement=False)
```

Comment:
Lets add it in the `kms_api()` transform utility as we are transforming this key for every operation in this service.

Reply:
`kms_api` will transform the `key_id` in either case, the regex transformer is used to preserve the structure of the field. Thats why [here](https://github.com/localstack/localstack/pull/8302/files#diff-b64eb19535de434dd14ad306c91e150306f94aa9a23a127046463b1f43d7bb95R1432) the key id is key arn.

Reply:
kms_api will transform the key_id in either case, the regex transformer is used to preserve the structure of the field. Thats why [here](https://github.com/localstack/localstack/pull/8302/files#diff-b64eb19535de434dd14ad306c91e150306f94aa9a23a127046463b1f43d7bb95R1432) the key id is key arn.

Reply:
Discussed in slack, will put it here for everyone: The tests were moved to a new class to avoid using the autouse fixture with the kms_api transformer.  This was done to avoid a blind key_value replacement of the `KeyId` field, which would have made it impossible to make assertions on its structure.